scripts/ensure-venv.sh

```diff
@@ -2,40 +2,42 @@
 
 # optionally opt out of virtualenv creation
 # WARNING: this will be removed (most likely renamed) soon!
-if [ "$SENTRY_NO_VIRTUALENV_CREATION" == "1" ]; then
+if [[ "$SENTRY_NO_VIRTUALENV_CREATION" == "1" ]]; then
     exit 0
 fi
 
 venv_name=".venv"
-python_bin="python2.7"
+python_version="3.6"
 
-if [ "$SENTRY_PYTHON3" = "1" ]; then
-    venv_name=".venv3"
-    python_bin="python3.6"
+if [[ "$SENTRY_PYTHON2" = "1" ]]; then
+    venv_name=".venv2"
+    python_version="2.7"
 fi
 
-die() {
-    cat >&2 "${@}"
+die () {
+    cat <<EOF
+$@
+EOF
     exit 1
 }
 
-if [ -n "$VIRTUAL_ENV" ]; then
-    # we're enforcing that virtualenv be in .venv, since future tooling e.g. venv-update will rely on this.
-    if [ "$VIRTUAL_ENV" != "${PWD}/${venv_name}" ]; then
-        die "You're in a virtualenv, but it's not in the expected location (${PWD}/${venv_name})"
-    fi
-
-    # TODO: Update this to strictly check .python-version
-    if [ "$SENTRY_PYTHON3" = "1" ]; then
-        python -c "import sys; sys.exit(sys.version_info[:2] != (3, 6))" ||
-            die "For some reason, the virtualenv isn't Python 3.6."
+if [[ -n "$VIRTUAL_ENV" ]]; then
+    major=`python -c "import sys; print(sys.version_info[0])"`
+    minor=`python -c "import sys; print(sys.version_info[1])"`
+    # If .venv contains Python2 and no SENTRY_PYTHON2 is set, then fail with instructions
+    if [[ major -eq 2 ]]; then
+        [[ -n "$SENTRY_PYTHON2" ]] ||
+            die \
```

Comment:
I thought we were going to keep the environment working? So should we not continue and just update the `python_version`?

Reply:
I have addressed all comments on the PR.

What about this?
- To use your existing .venv with Python simply set SENTRY_PYTHON2 and re-run
- To set up Python 3, run the following: source ./scripts/bootstrap-py3-venv
- To keep using Python 2, run: source ./scripts/bootstrap-py2-venv

I think I need to make the first option evident in the choices.

@evanh add the first option explicitly and call it done?

Reply:
No, if .venv is py2 then it should just continue to work, but with warnings.

Reply:
Yeah +1 for having everything just continue to work once we merge this